dotnet/src/IntegrationTests/Connectors/AzureOpenAI/AzureOpenAIChatCompletion_AutoFunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,379 @@
+    }
+
+    [Fact]
+    public async Task SpecifiedInCodeInstructsConnectorToInvokeNonKernelFunctionManuallyForStreamingAsync()
+    {
+        // Arrange
+        var plugin = this._kernel.CreatePluginFromType<DateTimeUtils>(); // Creating plugin without importing it to the kernel.
+
+        var invokedFunctions = new List<string>();
+
+        this._autoFunctionInvocationFilter.RegisterFunctionInvocationHandler(async (context, next) =>
+        {
+            invokedFunctions.Add(context.Function.Name);
+            await next(context);
+        });
+
+        var functionsForManualInvocation = new List<string>();
+
+        var settings = new AzureOpenAIPromptExecutionSettings { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto([plugin.First()], autoInvoke: false) };
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("How many days until Christmas?");
+
+        // Act
+        await foreach (var content in this._chatCompletionService.GetStreamingChatMessageContentsAsync(chatHistory, settings, this._kernel))
+        {
+            if (content is OpenAIStreamingChatMessageContent openAIContent && openAIContent.ToolCallUpdates is { Count: > 0 } && !string.IsNullOrEmpty(openAIContent.ToolCallUpdates[0].FunctionName))
+            {
+                functionsForManualInvocation.Add(openAIContent.ToolCallUpdates[0].FunctionName);
+            }
+        }
+
+        // Assert
+        Assert.Single(functionsForManualInvocation);
+        Assert.Contains("DateTimeUtils-GetCurrentDate", functionsForManualInvocation);
+
+        Assert.Empty(invokedFunctions);
+    }
+
+    private Kernel InitializeKernel()
+    {
+        var azureOpenAIConfiguration = this._configuration.GetSection("AzureOpenAI").Get<AzureOpenAIConfiguration>();
+        Assert.NotNull(azureOpenAIConfiguration);
+        Assert.NotNull(azureOpenAIConfiguration.ChatDeploymentName);
+        Assert.NotNull(azureOpenAIConfiguration.ApiKey);
+        Assert.NotNull(azureOpenAIConfiguration.Endpoint);
+
+        var kernelBuilder = base.CreateKernelBuilder();
+
+        kernelBuilder.AddAzureOpenAIChatCompletion(
+            deploymentName: azureOpenAIConfiguration.ChatDeploymentName,
+            modelId: azureOpenAIConfiguration.ChatModelId,
+            endpoint: azureOpenAIConfiguration.Endpoint,
+            apiKey: azureOpenAIConfiguration.ApiKey);
+
+        return kernelBuilder.Build();
+    }
+
+    private readonly IConfigurationRoot _configuration = new ConfigurationBuilder()
+        .AddJsonFile(path: "testsettings.json", optional: false, reloadOnChange: true)
+        .AddJsonFile(path: "testsettings.development.json", optional: true, reloadOnChange: true)
+        .AddEnvironmentVariables()
+        .AddUserSecrets<AzureOpenAIChatCompletionTests>()
+        .Build();
+
+    /// <summary>
+    /// A plugin that returns the current time.
+    /// </summary>
+    public class DateTimeUtils
```

Comment:
> [!NOTE]
> Apply for others

If this one is used only in this Test class, move it as private sealed.

```suggestion
    private sealed class DateTimeUtils
```

Reply:
fixed